k8s/envoy-jwt/README.md

```diff
@@ -0,0 +1,329 @@
+      Audience: a.config.audience,
+   })
+   if err != nil {
+      return forbiddenResponse("PERMISSION_DENIED"), nil
+   }
+
+   ...
+
+   // Parse and validate token against fetched bundle from jwtSource.
+   _, err := jwtsvid.ParseAndValidate(token, a.config.jwtSource, []string{a.config.audience})
+
+   if err != nil {
+      return forbiddenResponse("PERMISSION_DENIED"), nil
+   }
+```
+Note: _`workloadapi` and `jwtsvid` are imported from `go-spiffe` library._
+
+## Update Deployments
+
+The `auth-helper` service enables Envoy to inject or validate authentication headers carrying a JWT-SVID as described [above](#about-envoy-jwt-auth-helper).
+
+This new service must be added as a sidecar and must be configured to communicate with the SPIRE Agent. This is achieved by mounting a volume to share the Unix domain socket the SPIRE Agent provides. A new second volume provides access to the configmap defined with the service configuration. The following code snippet, from containers section, describes these changes:
+
+```console
+   - name: auth-helper
+     image: envoy-jwt-auth-helper:1.0.0
+     imagePullPolicy: IfNotPresent
+     args:  ["-config", "/run/envoy-jwt-auth-helper/config/envoy-jwt-auth-helper.conf"]
+     ports:
+     - containerPort: 9010
+     volumeMounts:
+     - name: envoy-jwt-auth-helper-config
+       mountPath: "/run/envoy-jwt-auth-helper/config"
+       readOnly: true
+     - name: spire-agent-socket
+       mountPath: /run/spire/sockets
+       readOnly: true
+```
+
+The `spire-agent-socket` volume is already defined for the deployment, no need to add it again. The configmap `envoy-jwt-auth-helper-config` needs to be added into the _volumes_ section, like this:
+
+```console
+   - name: envoy-jwt-auth-helper-config
+     configMap:
+        name: be-envoy-jwt-auth-helper-config
+```
+
+## Add an External Authorization Filter
+
+Next, this setup requires an External Authorization Filter in the Envoy configuration that connects to the new service. This new HTTP filter calls the `auth-helper` service just added to the deployment:
+
+```console
+   http_filters:
+   - name: envoy.ext_authz
+   config:
+      grpc_service:
+         envoy_grpc:
+         cluster_name: ext-authz
+         timeout: 0.5s
+```
+
+And the corresponding cluster:
+
+``` console
+  - name: ext-authz
+    connect_timeout: 1s
+    type: strict_dns
+    http2_protocol_options: {}
+    hosts:
+      - socket_address:
+          address: 127.0.0.1
+          port_value: 9010
+```
+
+If you are curious, the complete configuration file is located at `k8s/backend/config/envoy.yaml`.
+
+## Apply the New Resource
+
+In order for the new configuration to take effect, the services need to be redeployed. Let's remove backend and frontend deployments to get them updated:
+
+```console
+   $ kubectl delete deployment backend
+   $ kubectl delete deployment frontend
+```
+
+Deploy the new resources using:
+
+```console
+   $ kubectl apply -k k8s/.
```

Comment:
Thinking about the repo (spire-tutorials/k8s/envoy-jwt/k8s), there are two k8s directories, so let's be more specific.
```suggestion
Ensure that the current working directory is `.../spire-tutorials/k8s/envoy-jwt` and deploy the new resources using:

```console
   $ kubectl apply -k k8s/.
```